df_chat/asgi/consumers.py

```diff
@@ -90,44 +123,31 @@ async def message_activity(self, message: dict, **kwargs):
 
     @message_activity.groups_for_signal
     def message_activity(self, instance: Message, **kwargs):
-        yield f"-room__{instance.room_user.room_id}"
+        yield f"-rooms__{instance.room_user.room.id}"
 
     @message_activity.groups_for_consumer
-    def message_activity(self, consumer, **kwargs):
-        yield f"-room__{consumer.room_id}"
-
-    @message_activity.serializer
-    def message_activity(self, instance: Message, action, **kwargs) -> dict:
-        """This will return the message serializer"""
-        return MessageSerializer(instance).data
+    def message_activity(self, consumer, room_pk: str, **kwargs):
+        yield f"-rooms__{room_pk}"
 
+    @database_sync_to_async
     def _resolve_is_me(self, message: dict):
         if not isinstance(message["is_me"], bool):
-            message["is_me"] = message["is_me"] == str(self.room_user.id)
-
-    @database_sync_to_async
-    def check_room(self):
-        self.room = Room.objects.filter_for_user(self.scope["user"]).get(
-            pk=self.room_id
-        )
+            # TODO: IMPROVEMENT: There should find a way to not run a database query here
+            message["is_me"] = self.user.roomuser_set.filter(
```

Comment:
Let's add `user_id` field to `MessageSerializer` 

https://github.com/djangoflow/django-df-chat/blob/668ab2d039f30b5e65eedc674c19ec2af573bf5d/df_chat/drf/serializers.py#L111

Then we can compare `self.user['id'] == message["user_id"]` to compute `is_me`

Reply:
Thanks for the suggestion. Done.